File: Streams.Common/Model/Project.cs
using Streams.Common.Model;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text;

namespace Streams.Common
{
    public class Project
    {
        public Project()
        {
            AppTasks = new List<AppTask>();
        }

        public int Id { get; set; }

        [Required]
        public string Name { get; set; }
        [MaxLength(6), MinLength(3)]
        public string ProjectKey { get; set; }
        public int? ProjectTypeId { get; set; }
        public int? ProductId { get; set; }
        public int? WorkstreamId { get; set; }
        public int? OrganisationId { get; set; }

        public bool Active { get; set; }
        public bool Completed { get; set; }
        public string Description { get; set; }
        public string Owner { get; set; }
        [Required]
        [Range(1,4)]
        public int Priority { get; set; }

        public DateTime? DeadlineDate { get; set; } = DateTime.Now.AddDays(10);
        public DateTime? StartDate { get; set; } = DateTime.Now;
        public DateTime? CompletedDate { get; set; } 

        public List<AppTask> AppTasks { get; set; }

        public DateTime CreatedDate { get; set; } = DateTime.Now;
        public Byte[] TimeStamp { get; set; }

    }
}

File: streams-material/src/app/_models/project.ts
// export interface IProject {
//     id: number;
//     name: string;
//     projectkey: string;
//     owner: string;
//     active: boolean;
//     priority: number;
//     projectTypeId: number;
//     started: Date;
//     completed: Date;
//     deadline: Date;
//   }

export interface IProject {
    id: number;
    name: string;
    projectKey: string;
    projectTypeId: number;
    productId?: number;
    workstreamId?: number;
    OrganisationId?: number;
    active: boolean;
    completed: boolean;
    description: string;
    owner: string;
    priority: number;
    deadlineDate?: Date;
    startDate?: Date;
    completedDate?: Date;
    createdDate: Date;
  }

  /* THIS IS THE PROJECT FIELD LIST FROM THE API 
  id
  name
  projectKey
  projectTypeId
  workstreamId
  OrganisationId
  active
  completed
  description
  owner
  priority
  deadlineDate
  startDate
  completedDate
  createdDate
  timeStamp
  */

